src/NUnitFramework/tests/TestUtilities/TestBuilder.cs

```diff
@@ -135,6 +135,29 @@ public static ITestResult RunTestSuite(TestSuite suite, object testObject)
             return work.Result;
         }
 
+        public static CompositeWorkItem GenerateWorkItem(TestSuite suite, object testObject, ParallelScope? parallelScope)
+        {
+            TestExecutionContext context = new TestExecutionContext();
+            context.TestObject = testObject;
+            if(parallelScope != null) context.ParallelScope = (ParallelScope)parallelScope;
```

Comment:
ParallelScope does not seem to be relevant at all here. Tests run by testbuilder, as here, are executed by telling the workitem to execute them directly. ParallelScope is only relevant if the tests are to be run by a parallel dispatcher.